File: Go/Day06.go
package main

import AH "./adventhelper"

func getMarker(s string, n int) int {
	runes := []rune(s)

	for i := 0; i < len(runes) - n; i++ {
		p := runes[i:i+n]

		hash := make(map[rune]bool)

		for _, r := range p {
			hash[r] = true
		}

		if len(hash) == n {
			return i + n
		}
	}

	return 0
}

func main() {
	space, _ := AH.ReadStrFile("../input/input06.txt")

	AH.PrintSoln(6, getMarker(space[0], 4), getMarker(space[0], 14))

	return
}

File: Rust/src/day06.rs
use crate::advent_helper::advent_helper::read_strs;
use crate::advent_helper::advent_helper::print_soln;
use itertools::Itertools;

fn get_marker (s: &String, n: usize) -> usize
{
	let t = s.as_bytes().windows(n).position(|v| v.to_vec().iter().unique().collect::<Vec<&u8>>().len() == n).unwrap();
	return t + n
}

pub fn run()
{
	let input = read_strs("../input/input06.txt").unwrap();

	print_soln(6, get_marker(&input[0], 4), get_marker(&input[0], 14));
	return;
}
